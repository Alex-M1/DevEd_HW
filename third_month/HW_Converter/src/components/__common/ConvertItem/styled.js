import styled from 'styled-components';

export const StConvertItem = styled.div`
  display: grid;
  grid-template-rows: 1fr 1fr;
  gap: 10px;
  width: 200px;
`;
