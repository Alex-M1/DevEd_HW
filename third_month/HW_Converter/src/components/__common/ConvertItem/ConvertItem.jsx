import React from 'react';
import PropTypes from 'prop-types';
import Dropdown from '../Dropdown';
import Input from '../Input';
import { StConvertItem } from './styled';

function ConvertItem({ inputValue, setInputValue, values, dropValue, onChange, readOnly }) {
  return (
    <StConvertItem>
      <Input value={inputValue} onChange={setInputValue} readOnly={readOnly} />
      <Dropdown values={values} dropValue={dropValue} onChange={onChange} />
    </StConvertItem>
  );
}

export default ConvertItem;

ConvertItem.propTypes = {
  values: PropTypes.arrayOf(PropTypes.string).isRequired,
  dropValue: PropTypes.string.isRequired,
  inputValue: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  setInputValue: PropTypes.func,
  readOnly: PropTypes.bool,
};

ConvertItem.defaultProps = {
  readOnly: false,
};
