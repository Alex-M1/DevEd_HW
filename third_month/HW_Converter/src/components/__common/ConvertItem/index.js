import ConvertItem from './ConvertItem';

export default ConvertItem;
