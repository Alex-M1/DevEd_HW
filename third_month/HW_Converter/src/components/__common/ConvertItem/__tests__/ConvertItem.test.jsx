import { mount, shallow } from 'enzyme';
import React from 'react';
import ConvertItem from '../ConvertItem';

describe('Form', () => {
  const props = {
    values: ['UAH', 'EUR', 'USD', 'RUB'],
    readOnly: false,
    onChange: jest.fn(),
    dropValue: 'UAH',
    inputValue: '',
    setInputValue: jest.fn(),
  };
  afterEach(() => {
    props.onChange.mockClear();
  });
  it('Should match snapshot', () => {
    const component = shallow(<ConvertItem {...props} />);
    expect(component.html()).toMatchSnapshot();
  });
  it('Should have StConvertItem', () => {
    const component = mount(<ConvertItem {...props} />);
    expect(component.find('styled__StConvertItem')).toHaveLength(1);
  });
  it('Should have Input', () => {
    const component = mount(<ConvertItem {...props} />);
    expect(component.find('Input')).toHaveLength(1);
  });
  it('Should have Dropdown', () => {
    const component = mount(<ConvertItem {...props} />);
    expect(component.find('Dropdown')).toHaveLength(1);
  });
  it('Should have call onChange', () => {
    const component = mount(<ConvertItem {...props} />);
    component.props().onChange();
    expect(component.props().onChange).toHaveBeenCalledTimes(1);
  });
  it('Should have call setInputValue', () => {
    const component = mount(<ConvertItem {...props} />);
    component.props().setInputValue();
    expect(component.props().setInputValue).toHaveBeenCalledTimes(1);
  });
});
