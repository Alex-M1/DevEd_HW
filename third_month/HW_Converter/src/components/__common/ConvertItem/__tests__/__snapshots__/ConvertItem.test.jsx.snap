// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Should match snapshot 1`] = `"<div class=\\"styled__StConvertItem-sc-cq8jb4-0 calHuE\\"><input value=\\"\\" class=\\"styled__StInput-sc-c3z9ep-0 feLlzU\\"/><select class=\\"styled__StSelect-sc-7l3rob-0\\"><option selected=\\"\\" value=\\"UAH\\" class=\\"styled__StOption-sc-7l3rob-1\\">UAH</option><option value=\\"EUR\\" class=\\"styled__StOption-sc-7l3rob-1\\">EUR</option><option value=\\"USD\\" class=\\"styled__StOption-sc-7l3rob-1\\">USD</option><option value=\\"RUB\\" class=\\"styled__StOption-sc-7l3rob-1\\">RUB</option></select></div>"`;
